video_encoding/files.py

```diff
@@ -39,8 +41,15 @@ def _get_video_info(self):
         if not hasattr(self, '_info_cache'):
             encoding_backend = get_backend()
             try:
-                path = os.path.abspath(self.path)
-            except AttributeError:
-                path = os.path.abspath(self.name)
-            self._info_cache = encoding_backend.get_media_info(path)
+                _path = getattr(self, 'path', self.name)
+            except NotImplementedError:
+                _path = self.name
+            try:
+                path = default_storage.path(_path)
+            except NotImplementedError:
+                path = default_storage.url(_path)
```

Comment:
As far as I know, it is possible to limit the access to a file and one has to use a special [signed URL](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-presigned-urls.html). Would this automatically use that special URL if required? Additionally, If we are working with external storages, it might be a good idea to download the file first into a temporary folder instead of downloading it every time we need to do something with that file. What do you think? Maybe `django-storages` or similar already provides a caching mechanism.

Reply:
Yes we use this on AWS, temporal access tokens, all managed under the hood by the S3 storage implementation.

We shouldn't have to do anything specific to S3 or other implementations really.

Downloading it every-time sounds a bit redundant. The videos are to be processed only once, so we won't be accessing them over and over again.